test/libsolidity/semanticTests/constructor/constructor_function_argument.sol

```diff
@@ -0,0 +1,8 @@
+// The IR of this test used to throw
+contract D {
+    constructor(function() external returns (uint)) {}
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor() -> FAILURE
```

Comment:
Why does this test fail?

Reply:
Because the type is wrong, and because it is expecting a function type as an argument. Actually not sure how to call the constructor here. The other test is better in that regard.

BTW, this line is actually not needed, but isoltest complains if everything is empty. This is only meant to test that the contract compiles via IR.

Reply:
I think function types are abi encoded as a bytes4. So perhaps `0x20, 0xabcdefgh` where the latter is some random bytes4 (may need zero padding to the left)?

Reply:
Let me see if I can write a better test.